src/server/game/Entities/Object/Object.h

```diff
@@ -762,6 +783,9 @@ class TC_GAME_API WorldObject : public Object, public WorldLocation
 
         ObjectGuid _privateObjectOwner;
 
+        std::unordered_map<ObjectGuid /* Seer */, ReplaceObjectInfo> _replacedObjects;
```

Comment:
It is super heavy to add two `std::unordered_maps` to every object in the world.

Reply:
what is the alternative? the original npc can replace multiple "to be despawned personal copies" from x players and the original npc can be replaced by x personal copies for each player

Reply:
@Seyden I think that a vector could be more efficient in this case, how many ReplaceObject can have a npc?, 20 30?

Reply:
In case that you need a map, may @Naios could tell you if the boost::flat_map is a good choice for this.

Reply:
@BertiRean the count is: max players on the server since start until creature respawn